File: week_8_and_9/4_Ruby/reverse_words/my_solution.rb
# U3.W8-9: Reverse Words


# I worked on this challenge [by myself].

# 2. Pseudocode



# 3. Initial Solution
=begin
def reverse_words(sentence)
	sentence.reverse
end

=end
# 4. Refactored Solution

def reverse_words(sentence)
	sentence.split(' ').map {|word| word.reverse}.join(' ')
end

# 1. DRIVER TESTS/ASSERT STATEMENTS GO BELOW THIS LINE

p reverse_words("Hello") == "olleH"

p reverse_words("Hakuna Matata") #== "anukaH atataM"

p reverse_words("") == ""

p reverse_words("How are you") #== "woH era uoy"



# 5. Reflection 
=begin
	
I found this to be a relatively easy challenge. My first solution was a little too simplified, and only worked when I wanted to reverse one 
word or an empty string. My final solution, I just had to loop through the sentence to reverse each word separately. I didn't have to research
any methods for this challenge, but I did have to take a look into others solutions to remember to use the split and join method to make the 
solution work exactly right.
	
=end

File: week_8_and_9/3_JavaScript/translate_to_js/my_solution.js
// U3.W8-9: Challenge you're converting
// Nums to Words Challenge

// I worked on this challenge [by myself].

// 2. Pseudocode

// Define a variable, super_fizzbuzz, that takes an array as the argument. 
// Use a for loop that:
// 	if x%15 == 0 return FizzBuzz for x
// 	else if x%3 == 0 return Fizz for x
// 	else if x%5 == 0 return Buzz for x
//  else returns x


// 3. Initial Solution

// var super_fizzbuzz = function(array) {
// 	array.map(function(num) {
// 		if (num% 15==0) {
// 			console.log('FizzBuzz');}
// 		else if (num%5 == 0) {
// 			console.log('Buzz');}
// 		else if (num%3 == 0) {
// 			console.log('Fizz');}
// 		else {
// 			console.log(num);}
// 		})
// 	};



// 4. Refactored Solution

var super_fizzbuzz = function(array) {
	array.map(function(num) {
		console.log((num% 15===0 ? 'FizzBuzz':'') || (num%5 === 0 ? 'Buzz' : '') + (num%3 ===0 ? 'Fizz' : '') || num);
		})
	};





// 1. DRIVER TESTS/ASSERT STATEMENTS GO BELOW THIS LINE

super_fizzbuzz([1,3,5,15,9,4])

super_fizzbuzz([25,100,300,276,23654])

super_fizzbuzz([1])





// 5. Reflection 
// When I started this challenge, I had a good idea about
// how I wanted to solve it, but I wasn't sure how to do it in 
// JavaScript. I had to research how to loop through an array
// to access each element, but once I found that, I was able to
// find my solution. I found the refactored code on rosettacode.org, and then I just had to tweek it a little to suit 
// my purposes for this challenge. I enjoyed the challenge of turning the super_fizzbuzz method from ruby into javascript.
